src/events.ts

```diff
@@ -13,7 +13,12 @@ export interface CommonDeviceEvent<
 export type DeviceConnectedEvent = CommonDeviceEvent<"device.connected">
 export type DeviceDisconnectEvent = CommonDeviceEvent<"device.disconnected">
 export type DeviceTamperEvent = CommonDeviceEvent<"device.tampered">
-export type DeviceLowBatteryEvent = CommonDeviceEvent<"device.low_battery">
+export type DeviceLowBatteryEvent = CommonDeviceEvent<
+  "device.low_battery",
+  {
+    code: "hub_disconnected" | "device_disconnected"
```

Comment:
Sorry, probably just misunderstanding something--why do we want to attach disconnection codes to the low battery event?

Reply:
A potential approach for story mentioned in this issue: https://github.com/seamapi/seam-connect/issues/817

Reply:
i think a /device/diagnose_issue endpoint could take into consideration the low battery event and the disconnected event and give a likely reason. We cant often guarantee the reason for disconnection is low battery, so we shouldn't include it in the payload for disconnected.

Reply:
The disconnected event should have disconnection_type: "hub_disconnected" | "device_disconnected"

Reply:
😵‍💫  I'm actually re-reading not just the code, but also Max's comment, yeah, this was just a silly mistake, meant to go on the device.disconnected event, not on device.low_battery, will fix

Reply:
Sg! Also i think "error_code" over code might be better and todo we should define our list of error codes on the docs

Reply:
thanks @codetheweb for pointing this out, apologies for not seeing it even after you pointed it out, fixed now 😅